.github/workflows/build-image.yml

```diff
@@ -0,0 +1,115 @@
+name: Build test harness image
```

Comment:
I guess if this is extracted into it's own workflow already that we could make it live in a shared repo right?

It seems a bit odd that for example this takes an input of what image name to build when that's fixed/constant for this repo but I guess you're pointing out we could share this across all harness repos as it captures common logic. 

I'm good with doing this at a later step though as well.

Reply:
Yes, you are absolutely right @Julian. Actually, I was thinking to move both workflows (one that builds all images for available versions and the PR workflow) and reuse them across all the test harnesses when we move them